### PR TITLE
Cwdoe 1382 msel injects

### PR DESCRIPTION
**Pull request by sei-tspencer**
added create from inject endpoint
added list of data fields to catalog
added isNew and DisplayOrder to CatalogInject

### CODE REVIEW

File: Blueprint.Api/Services/ScenarioEventService.cs [hunk 1]
```diff
@@ -141,6 +144,128 @@ public async Task<IEnumerable<ViewModels.ScenarioEvent>> CreateAsync(ViewModels.
...
+                    {
+                        Id = Guid.NewGuid(),
+                        MselId = msel.Id,
+                        Name = injectDataField.Name,
+                        DataType = injectDataField.DataType,
+                        DisplayOrder = ++displayOrder,
+                        DateCreated = dateCreated,
+                        CreatedBy = userId,
+                        IsChosenFromList = injectDataField.IsChosenFromList,
+                        OnScenarioEventList = showIt,
```
Comment (sei-mpriest) [inline Blueprint.Api/Services/ScenarioEventService.cs:195]: Wouldn't you want to get this from the inject DataField if it is there and its not an html type?

File: Blueprint.Api.Migrations.PostgreSQL/Migrations/20240717193150_list-datafields.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#nullable disable
+
+namespace Blueprint.Api.Migrations.PostgreSQL.Migrations
+{
+    public partial class listdatafields : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "list_data_fields",
```
Comment (sei-mpriest) [inline Blueprint.Api.Migrations.PostgreSQL/Migrations/20240717193150_list-datafields.cs:17]: I see that we are passing this list from the ui, but I don't understand it's significance. What does the list mean?

File: Blueprint.Api.Data/Models/CatalogInject.cs [hunk 3]
```diff
@@ -28,6 +28,8 @@ public CatalogInjectEntity(Guid injectId, Guid catalogId)
 
         public Guid CatalogId { get; set; }
         public CatalogEntity Catalog { get; set; }
+        public bool IsNew { get; set; }
```
Comment (sei-mpriest) [inline Blueprint.Api.Data/Models/CatalogInject.cs:31]: How does something become not new? It seems like this means its been added to a MSEL, but I don't see how it gets updated.

File: Blueprint.Api/Services/ScenarioEventService.cs [hunk 4]
```diff
@@ -141,6 +144,128 @@ public async Task<IEnumerable<ViewModels.ScenarioEvent>> CreateAsync(ViewModels.
...
+            var dateCreated = DateTime.UtcNow;
+            // create a data field dictionary and create any missing data fields
+            var dataFieldDictionary = new Dictionary<Guid, Guid>();
+            var displayOrder = msel.DataFields.Count > 0
+                ? msel.DataFields.Max(m => m.DisplayOrder)
+                : 0;
+            foreach (var injectDataField in injectType.DataFields)
+            {
+                var showIt = injectDataField.DataType != DataFieldType.Html;
+                var mselDataFieldEntity = msel.DataFields.SingleOrDefault(m => m.Name == injectDataField.Name && m.DataType == injectDataField.DataType);
```
Comment (sei-mpriest) [inline Blueprint.Api/Services/ScenarioEventService.cs:182]: Oh, is this where you are supposed to see if the field is in the InjectIdList? Is that what that field is for: to specify which fields from the template to actually add?

File: Blueprint.Api.Data/Models/CatalogInject.cs (hunk 3 above)
Comment (sei-tspencer) [reply Blueprint.Api.Data/Models/CatalogInject.cs:31]: "New" is user data and means that it is a new catalog item.  The creator of the catalog decides which injects are "new" and which ones are not.  It is edited on the catalog edit dialog in the UI.

File: Blueprint.Api.Migrations.PostgreSQL/Migrations/20240717193150_list-datafields.cs (hunk 2 above)
Comment (sei-tspencer) [reply Blueprint.Api.Migrations.PostgreSQL/Migrations/20240717193150_list-datafields.cs:17]: These are the DataFields that are actually displayed on the catalog inject list when displaying the list of injects contained in a catalog.

File: Blueprint.Api/Services/ScenarioEventService.cs (hunk 4 above)
Comment (sei-tspencer) [reply Blueprint.Api/Services/ScenarioEventService.cs:182]: This is where we check to see which of the DataFields that are on the Inject are not already on the MSEL.  If it is not already on the MSEL, we add it to the MSEL.

**Review by sei-mpriest — APPROVED** [review]

**Pull request merged**